sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -604,7 +604,7 @@ class Analyzer(
 
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
-        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
+        i.copy(table = resolveRelation(EliminateSubqueryAliases(lookupTableFromCatalog(u))))
```

Comment:
When we try to insert into a view, the logical plan that `lookupTableFromCatalog()` returns is not expanded, so we have to perform `resolveRelation()` over the node.